src/test/ui/issues/issue-49824.nll.stderr

```diff
@@ -1,17 +1,15 @@
-error: unsatisfied lifetime constraints
+error: captured variable cannot escape `FnMut` closure body
   --> $DIR/issue-49824.rs:22:9
    |
 LL |       || {
-   |       --
-   |       ||
-   |       |return type of closure is [closure@$DIR/issue-49824.rs:22:9: 24:10 x:&'2 mut i32]
-   |       lifetime `'1` represents this closure's body
+   |        - inferred to be a `FnMut` closure
 LL | /         || {
 LL | |             let _y = &mut x;
 LL | |         }
```

Comment:
In this error, it's not clear that something is being returned (to me), yet the category that we're getting for this is `ConstraintCategory::Return`, therefore it's taking this code branch - I'm not sure that this error is better for this case. 

That said, this part of the previous error isn't great:

```
|return type of closure is [closure@$DIR/issue-49824.rs:22:9: 24:10 x:&'2 mut i32]
```

Reply:
The outer closure is returning the inner one; isn’t that what the category is reflecting?

Reply:
I don't know if the right route is to remove the closure type entirely.

The printout here: `|return type of closure is [closure@$DIR/issue-49824.rs:22:9: 24:10 x:&'2 mut i32]` tells us that the outer closure is returning a closure, and then the printout describes that closure, in terms of the closure's span and the record structure representing the *captured free variable(s)* of the closure, and the types of those variables.

Should we consider trying to render that information a bit more legibly, instead of removing it entirely?

Reply:
Ah, I hadn't realised it was the outer closure returning, that makes a bit more sense. I think in general, we should render the closure type in a more legible way, though I have no idea whether we should in this case or not.